File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        longestSequence = 0
        hashset = set(nums)

        for num in hashset:
            if (num - 1) not in hashset:
                length = 1
                while (num + length) in hashset:
                    length += 1
                longestSequence = max(length, longestSequence)
        
        return longestSequence

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const count = {};
        let freq = Array.from({ length: nums.length + 1 }, () => []);

        for (let num of nums){
            count[num] = 1 + (count[num] || 0);
        }

        for (let i in count) {
            freq[count[i]].push(Number(i));
        }

        let ans = [];

        for (let i = freq.length-1; i > 0; i--) {
            for (let j of freq[i]) {
                ans.push(j);
                if (ans.length === k){
                    return ans;
                }
            }
        }
    }
}
